src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -1913,24 +1913,23 @@ let CheckRecdExprDuplicateFields (elems: Ident list) =
 //-------------------------------------------------------------------------
 
 /// Helper used to check record expressions and record patterns
-let BuildFieldMap (cenv: cenv) env isPartial ty (flds: ((Ident list * Ident) * 'T) list) m =
+let BuildFieldMap (cenv: cenv) env isPartial ty (flds: ExplicitOrSpread<(Ident list * Ident) * 'Explicit, (Ident list * Ident) * 'Spread> list) m =
```

Comment:
This file has grown by a lot and pieces of the logic are fairly spread specific.
Can you imagine extracting the biggest blocks into a separate file, with passing in mutually resursive functions as a callback (like TcExprFlex) ?